packages/svelte2tsx/src/emitDts.ts

```diff
@@ -13,7 +13,13 @@ export async function emitDts(config: EmitDtsConfig) {
     const { options, filenames } = loadTsconfig(config, svelteMap);
     const host = await createTsCompilerHost(options, svelteMap);
     const program = ts.createProgram(filenames, options, host);
-    program.emit();
+
+    return program.emit(undefined, (...args) => {
+        const [filename] = args;
+        if (isSubpath(config.declarationDir, path.resolve(filename))) {
+            host.writeFile(...args);
```

Comment:
Can you add a little comment explaining why we're doing this?

Reply:
Done